src/main/kotlin/com/kroffle/knitting/config/R2dbcConfiguration.kt

```diff
@@ -0,0 +1,33 @@
+package com.kroffle.knitting.config
+
+import com.kroffle.knitting.AppProperties
+import com.kroffle.knitting.domain.DesignType
+import com.kroffle.knitting.domain.PatternType
+import io.r2dbc.postgresql.PostgresqlConnectionConfiguration
+import io.r2dbc.postgresql.PostgresqlConnectionFactory
+import io.r2dbc.postgresql.codec.EnumCodec
+import io.r2dbc.spi.ConnectionFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.data.r2dbc.config.AbstractR2dbcConfiguration
+import org.springframework.data.r2dbc.repository.config.EnableR2dbcRepositories
+import org.springframework.beans.factory.annotation.Autowired
+
+@Configuration
+@EnableR2dbcRepositories
+class R2dbcConfiguration: AbstractR2dbcConfiguration() {
+    @Autowired
+    lateinit var appProperties: AppProperties
+
+    @Bean
+    override fun connectionFactory(): ConnectionFactory = PostgresqlConnectionFactory(
+        PostgresqlConnectionConfiguration.builder()
+            .host(appProperties.host)
+            .username(appProperties.username)
+            .password(appProperties.password)
+            .database(appProperties.database)
+            .codecRegistrar(EnumCodec.builder().withEnum("design_type", DesignType::class.java).build())
+            .codecRegistrar(EnumCodec.builder().withEnum("pattern_type", PatternType::class.java).build())
+            .build()
+    )
```

Comment:
design_type, pattern_type 같은 것들은 상수로 다루면 좋을 것 같아요! (돌려보진 않았어요)

```suggestion
            .codecRegistrar(EnumCodec.builder().withEnum(DESIGN_TYPE, DesignType::class.java).build())
            .codecRegistrar(EnumCodec.builder().withEnum(PATTERN_TYPE, PatternType::class.java).build())
            .build()
    )

    companion object {
        private const val DESIGN_TYPE = "design_type"
        private const val PATTERN_TYPE = "pattern_type"
    }
```

Reply:
왜 companion object 안에 정의해야하는지 궁금해서 찾아봤는데, https://stackoverflow.com/a/50520935 그냥 object 안에 선언하라는 의견이 있습니다!
혹시 왜 companion object 안에 정의하는 것을 제안해주셨는지 알 수 있을까요? 추가로 그냥 object 안에 선언하는 것은 어떨까요?

Reply:
그냥 object로 하면 이름을 정의해줘야하군요 🤔  일단 두리님이 제안해주신 방법으로 반영해두었습니다! https://github.com/k-roffle/knitting-service/pull/5/commits/3d88d9365c6a629cd4eda9f1dfdcb2becfad67bb

Reply:
design_type, pattern_type 같은 경우에는 상수로 관리하는 게 좋을 것 같다고 판단했고, 코틀린에서 상수는 companion object로 주로 관리하는 걸로 알고 있어서 제안드렸습니다!! 반영해주셔서 감사합니다! 앞으로 제안 코멘트는 좀 더 자세히 적어놓을게요~!

Reply:
companion object는 클래스 맨 하단에 정의하는 걸로 알고 있습니다!

Reply:
위에 코멘트에서 남긴 링크보면 코틀린에서 상수를 정의 하는 방법에 대한 질문에 가장 많은 vote를 받은 답변이 companion object를 기피하라는 답변인데요, 혹시 괜찮은거 맞을지 확인해주실 수 있을까요?

> companion object는 클래스 맨 하단에 정의하는 걸로 알고 있습니다! 요 부분만 한 번 더 봐주세요~!

이건 언어의 스타일 관례로써 하단에 정의하는걸까요? 돌아가는데 문제가 없고 찾아봤을 때 관련된 내용을 확인할 수 없어서 제가 참고할만한 문서가 있다면 알려주시면 좋을 것 같습니다!

Reply:
> 위에 코멘트에서 남긴 링크보면 코틀린에서 상수를 정의 하는 방법에 대한 질문에 가장 많은 vote를 받은 답변이 companion object를 기피하라는 답변인데요, 혹시 괜찮은거 맞을지 확인해주실 수 있을까요?

그렇군요.. 🤔 저도 위에 작성한 내용 만큼만 알고 있어서 그런 문제가 있다면 다른 방식도 괜찮을 것 같습니다!

> 이건 언어의 스타일 관례로써 하단에 정의하는걸까요? 돌아가는데 문제가 없고 찾아봤을 때 관련된 내용을 확인할 수 없어서 제가 참고할만한 문서가 있다면 알려주시면 좋을 것 같습니다!

넵! 돌아가는데에는 문제 없고 관례 중 하나입니다! 코틀린 공식 문서에도 있었던 것 같은데 저도 지금 찾으려니까 잘 못찾겠네요 ㅠㅜ 린트에서 안 걸린거면 크게 문제는 없을 것 같네요!! 참고만 해주세요!

Reply:
아하 그렇군요! 린트는 따로 깨지지 않는데, 이 PR 베이스로 하는 PR이 많아서 일단 머지하고 다음 PR에서 고치도록 하겠습니다! 설명 감사해요!